packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -1010,13 +1012,31 @@ export function TransferPanel() {
   }
 
   // Only run gas estimation when it makes sense, i.e. when there is enough funds
-  const shouldRunGasEstimation = useMemo(
-    () =>
-      isDepositMode
-        ? Number(amount) <= Number(l1Balance)
-        : Number(amount) <= Number(l2Balance),
-    [isDepositMode, l1Balance, amount, l2Balance]
-  )
+  const shouldRunGasEstimation = useMemo(() => {
```

Comment:
Because we got rid of `l1Balance` and `l2Balance` that changed based on whether you had ETH or an ERC-20 selected, now we gotta do some extra logic, but it's easier to understand what's going on IMO.